File: Easy/1716. Calculate Money in Leetcode Bank.java
class Solution {
    public int totalMoney(int days) {
        int currentWeek = 1;
        int prefixSum = 0;
        for (int i = 1; i < 8; i++) {
            prefixSum += i;
        }
        int totalMoney = 0;
        while (days / 7 > 0) {
            totalMoney += prefixSum;
            prefixSum += 7;
            currentWeek += 1;
            days -= 7;
        }
        for (int i = currentWeek; i < currentWeek + days; i++) {
            totalMoney += i;
        }
        return totalMoney;
    }
}

File: Easy/1716. Calculate Money in Leetcode Bank.py
weekly = 7 * 8 // 2
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks = n // 7
        n %= 7
        return weekly * weeks + ((weeks - 1) * weeks) // 2 * 7 + weeks * n + (n * (n + 1)) // 2